components/calculation/MeasureHighlightingPanel.tsx

```diff
@@ -21,24 +23,25 @@ export interface MeasureHighlightingPanelProps {
 export default function MeasureHighlightingPanel({ dr }: MeasureHighlightingPanelProps) {
   const { classes } = useStyles();
   const [searchValue, setSearchValue] = useState('');
+  const measureBundle = useRecoilValue(measureBundleState);
+  const measure = useMemo(() => {
+    return measureBundle.content?.entry?.find(e => e.resource?.resourceType === 'Measure')?.resource as fhir4.Measure;
+  }, [measureBundle]);
 
-  const parsedHTML = parse(dr.html || '', {
-    replace: elem => {
-      if ((elem as DomElement).attribs?.['data-statement-name']) {
-        const statementName = (elem as DomElement).attribs['data-statement-name'];
-        const libraryName = (elem as DomElement).attribs['data-library-name'];
-        const statementResult = dr.statementResults.find(
-          statement => statement.statementName === statementName && statement.libraryName === libraryName
-        );
+  const parsedHTML = () => {
+    const relevantStatements = dr.statementResults.filter(s => s.relevance !== Relevance.NA);
+    sortStatements(measure, dr.groupId, relevantStatements);
+    return relevantStatements.map(sr => {
+      if (sr.statementLevelHTML) {
         return (
           <>
-            {domToReact([elem])}
-            <PrettyOutput statement={statementResult} />
+            {parse(sr.statementLevelHTML)}
+            <PrettyOutput statement={sr} />
```

Comment:
It looks like I'm getting a unique key error 
<img width="648" alt="Screenshot 2025-04-16 at 7 42 31 AM" src="https://github.com/user-attachments/assets/cadf5db8-ae53-4ca8-b379-42b319cf78fe" />
that is from here. wrapping it with some unique key should fix it something like:
``` 
const key = `${dr.groupId}-${sr.libraryName}-${sr.statementName}`;
<div key={key}>
        {parse(sr.statementLevelHTML)}
        <PrettyOutput statement={sr} />
      </div>

Reply:
Good catch!!